repository.py

```diff
@@ -208,6 +235,22 @@ def load_charm(
     except OSError:
         raise RepositoryError(f"Failed to read file `{charm / CHARMCRAFT_FILE}`.")
 
+    deps = set()
+    pending = [charm.name]
+    while pending:
+        package = pending.pop()
+        deps.add(package)
+        for pkg_dep in next(
+            (pkg.get("dependencies", []) for pkg in uv_lock["package"] if pkg["name"] == package)
+        ):
+            if pkg_dep["name"] not in deps:
+                deps.add(pkg_dep["name"])
+                pending.append(pkg_dep["name"])
```

Comment:
Could you add a comment here or docstring to the `load_charm` function that explains why we need this recursive while loop? I know it's because `uv` doesn't resolve the dependencies of dependencies, but I could totally see someone without our context coming in and being like "wahhh :thinking:" 

We'll be thanking ourselves in six months if we need to rework this code at one point and we're trying to remember we're running this loop to analyze uv.lock files.